public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -144,6 +145,8 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
       keybindingSrv: this.context.keybindings,
     });
 
+    // local storage to handle last used datasource
+    initLocalStorageDashboard(match.params.uid);
```

Comment:
If we're relying on the URL params, why don't we use the location directly next to the component that is saving the dashboard UID?